sqlx-core/src/testing/mod.rs

```diff
@@ -41,6 +43,17 @@ pub trait TestSupport: Database {
     /// This snapshot can then be used to generate test fixtures.
     fn snapshot(conn: &mut Self::Connection)
         -> BoxFuture<'_, Result<FixtureSnapshot<Self>, Error>>;
+
+    /// Generate a unique database name for the given test path.
+    fn db_name(args: &TestArgs) -> String {
+        let mut hasher = Sha512::new();
+        hasher.update(args.test_path.as_bytes());
+        let hash = hasher.finalize();
+        let hash = URL_SAFE.encode(&hash[..39]);
```

Comment:
I think it's important to use `-` to `_` here, since `-` is not a valid character for SQL identifiers.

```rs
let db_name = format!("_sqlx_test_{}", hash).replace("-", "_");
```

I've had errors related to this